transform/app/transform/utils/common.py

```diff
@@ -629,6 +644,75 @@ def add_tg_fields(df: DataFrame) -> DataFrame:
     return df
 
 
+def harvest_exportation(df: DataFrame, harvested_properties: dict) -> None:
+    """
+    Harvest exportation information from instances within the DataFrame
+    Args:
+        df (DataFrame): Input DataFrame containing instance information.
+        harvested_properties (dict): Dictionary to store harvested properties.
+    Assumptions:
+        - Only the first 10 versions of each instance are harvested; subsequent versions are skipped
+          (approx. 0.2% of data is skipped).
+    For each instance:
+    - Extracted Fields:
+        - URL: URL of the instance.
+        - Type: Type of the instance.
+        - Publication Year: Year of publication from the publication date.
+        - License: License information.
+        - PIDs: Persistent identifiers associated with the instance.
+        - Hosted By: The entity hosting the instance.
+    The extracted information is structured into a list of dictionaries for each instance and stored in
+    'harvested_properties[EXPORTATION]'.
+    Note:
+    - 'i' is used to limit harvesting to the first 10 versions of each instance.
+    """
+    instances_list = df.select(INSTANCE).collect()
+    exportation_column = []
+
+    for instances in instances_list:
+        if instances[INSTANCE]:
+            exportation_row = []
+
+            for i, instance in enumerate(instances[INSTANCE]):
+                if i >= 9:
```

Comment:
Few issues here:
1) Define the limit somewhere explicitly.
E.g. `INSTANCES_LIMIT = 10`
Then `if i >= 9:` wouldn't be such a magical number.
It will become `if i > INSTANCES_LIMIT`:

2)  Please avoid using names such as "i" - its not that informative. In that case, I would use something like `instance_idx`

Reply:
You're correct, I've set the variable `instances_limit = 10` and renamed `i` to `instance_idx`. Moreover, I included `start=1` in `enumerate` to ensure an accurate limit of 10.